### PR TITLE
FreeBSD support

### PR DESCRIPTION
**Pull request by MuhammadMouradG**
### Describe your changes in detail
This PR add support to `Freebsd13`. As the `Freebsd` is an `unix-like` OS and could use `gtk`, it can be used with `toga` with just few edits to the `toga` code base.

For consistency, this PR changed the `linux` backend name to `Unix-like`, which make sense to any platform that is not `linux` but using `gtk3`.

### What problem does this change solve?
This PR make `Toga` possible on `Freebsd13` :smile:.

## PR Checklist:
- [x] All new features have been tested
- [x] All new features have been documented
- [x] I have read the **CONTRIBUTING.md** file
- [x] I will abide by the code of conduct

### CODE REVIEW

File: changes/1836.feature.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Add support to Freebsd13 OS.
```
Comment (freakboy3742) [inline changes/1836.feature.rst:1]: We try really hard to follow style guides when referring to other operating systems in user-facing text - I'm pretty sure this should be `FreeBSD`. Also - given FreeBSD12 isn't EOL, and FreeBSD14 is on the horizon - is there any reason to narrow this to v13 specifically?

File: core/setup.cfg [hunk 2]
```diff
@@ -45,6 +45,8 @@ keywords =
     android
     windows
     winforms
+	unix-like
+	freebsd
```
Comment (freakboy3742) [inline core/setup.cfg:49]: Indentation is inconsistent here. Also "Unix-like" is a near meaningless term. Android and macOS are both "unix like".

File: core/src/toga/platform.py [hunk 3]
```diff
@@ -20,7 +20,8 @@
     "android": "android",
     "darwin": "macOS",
     "ios": "iOS",
-    "linux": "linux",
+    "linux": "UnixLike",
+    "freebsd13": "UnixLike",
```
Comment (freakboy3742) [inline core/src/toga/platform.py:24]: This breaks backwards compatibility for existing code that is expecting "linux" as the toga platform.

The right approach here is to add a mapping for `freebsd13` to `freebsd` (or, to `freebsd13` if you think there's a need to differentiate versions in a Toga context; I can't say I know FreeBSD well enough to comment on whether this is the case). Adding mappings for 12 and 14 would also make sense.

File: docs/how-to/contribute-code.rst [hunk 4]
```diff
@@ -31,7 +31,7 @@ First thing is to ensure that you have Python 3 and pip installed. To do this ru
       $ python3 --version
       $ pip3 --version
 
-  .. group-tab:: Linux
+  .. group-tab:: Unix-like
```
Comment (freakboy3742) [inline docs/how-to/contribute-code.rst:34]: Unix-like is a near meaningless term, and one that only serves to confuse inexperienced Linux users. FreeBSD use is sufficiently niche that I'm comfortable expecting FreeBSD users to work out that their commands are close enough to Linux that they can adapt.

File: docs/how-to/contribute-code.rst [hunk 5]
```diff
@@ -82,10 +82,14 @@ Next, install any additional dependencies for your operating system:
...
     No additional dependencies
 
-  .. group-tab:: Linux
+  .. group-tab:: Unix-like
 
     .. code-block:: bash
 
+      # Freebsd13
+      (venv) $ sudo pkg update
+      (venv) $ sudo pkg install gtk3 gobject-introspection cairo webkit2-gtk3
```
Comment (freakboy3742) [inline docs/how-to/contribute-code.rst:91]: 1. `FreeBSD`, not `Freebsd`
2. Is there any reason to believe this set of instructions won't also be true for FreeBSD12 and 14? Could this just be `FreeBSD`, not `FreeBSD 13`
3. From a UX perspective, FreeBSD should be *last* in this list of instructions, not first. 
4. This list should also include anything needed to install a version of Python that can be used for development purposes.

File: docs/tutorial/tutorial-0.rst [hunk 6]
```diff
@@ -71,6 +71,10 @@ Next, install Toga into your virtual environment:
 
     .. code-block:: bash
 
+      # Freebsd13
+      (venv) $ sudo pkg update
+      (venv) $ sudo pkg install gtk3 gobject-introspection cairo webkit2-gtk3
```
Comment (freakboy3742) [inline docs/tutorial/tutorial-0.rst:76]: See notes above about these install instructions.

File: gtk/setup.cfg [hunk 7]
```diff
@@ -38,6 +38,8 @@ keywords =
     cross-platform
     toga
     desktop
+	unix-like
+	freebsd
```
Comment (freakboy3742) [inline gtk/setup.cfg:42]: See previous notes about indentation and `unix-like`

File: gtk/setup.cfg [hunk 8]
```diff
@@ -50,7 +52,7 @@ zip_safe = False
 
 [options.entry_points]
 toga.backends =
-    linux = toga_gtk
+    UnixLike = toga_gtk
```
Comment (freakboy3742) [inline gtk/setup.cfg:55]: Registering a backend costs nothing; this should 2 independent `linux` and `freebsd` registrations.

File: toga/setup.cfg [hunk 9]
```diff
@@ -45,6 +45,8 @@ keywords =
     android
     windows
     winforms
+	Unix-like
+	freebsd
```
Comment (freakboy3742) [inline toga/setup.cfg:49]: See previous comments.

File: toga/setup.py [hunk 10]
```diff
@@ -8,6 +8,7 @@
     extras_require={
         # Automatically installed platform backends
         ':sys_platform=="win32"': ["toga-winforms==%s" % version],
+        ':sys_platform=="freebsd13"': ["toga-gtk==%s" % version],
```
Comment (freakboy3742) [inline toga/setup.py:11]: Compatibility with 12 and 14 again.

**Review by freakboy3742 — CHANGES_REQUESTED** [review]
I don't have any objection adding the internal tooling to support use of FreeBSD; however, I'm not going to degrade the user experience for the larger Linux user base to accomodate it. Comments inline.

File: changes/1836.feature.rst (hunk 1 above)
Comment (MuhammadMouradG) [reply changes/1836.feature.rst:1]: Oh, you are right, this should be `FreeBSD`.

The main reason is that I'm tested it on v13, and I don't want to add version that I didn't test it. Also, I don't want to see, in the future, a list of 'How to install Toga in different `FreeBSD` versions', so I preferred to maintaining only last `FreeBSD` release, especially since `FreeBSD` is a rolling release OS. However, I suggest to remove the version returned by the `sys.platform` to be just `freebsd`.

File: core/src/toga/platform.py (hunk 3 above)
Comment (MuhammadMouradG) [reply core/src/toga/platform.py:24]: The changes addressed into the following commits, with removing the `freebsd` version.

File: README.rst [hunk 11]
```diff
@@ -41,9 +41,9 @@ Minimum requirements
 
 * If you're on macOS, you need to be on 10.10 (Yosemite) or newer.
 
-* If you're on Linux, you need to have GTK+ 3.10 or newer. This is the version
-  that ships starting with Ubuntu 14.04 and Fedora 20. You also need to install
-  the system packages listed in `Tutorial 0 <docs/tutorial/tutorial-0.rst>`__.
+* If you're on Unix-like, you need to have GTK+ 3.10 or newer. This is the
+  version that ships starting with Ubuntu 14.04 and Fedora 20. You also need to
+  install the system packages listed in `Tutorial 0 <docs/tutorial/tutorial-0.rst>`__.
```
Comment (freakboy3742) [inline README.rst:46]:
```suggestion
* If you're on Linux (or another Unix-based operating system), you need to have
  GTK+ 3.10 or newer. This is the version that ships starting with Ubuntu 14.04
  and Fedora 20. You also need to install the system packages listed in
  `Tutorial 0 <docs/tutorial/tutorial-0.rst>`__.```

File: changes/1836.feature.rst [hunk 12]
```diff
@@ -0,0 +1 @@
+Add support to FreeBSD OS.
```
Comment (freakboy3742) [inline changes/1836.feature.rst:1]:
```suggestion
Support for FreeBSD was added.
```

File: core/src/toga/platform.py [hunk 13]
```diff
@@ -37,7 +38,10 @@
     if hasattr(sys, "getandroidapilevel"):
         current_platform = "android"
     else:
-        current_platform = _TOGA_PLATFORMS.get(sys.platform)
+        if sys.platform.startswith("freebsd"):
+            current_platform = "freeBSD"
+        else:
+            current_platform = _TOGA_PLATFORMS.get(sys.platform)
```
Comment (freakboy3742) [inline core/src/toga/platform.py:44]:
```suggestion
    elif sys.platform.startswith("freebsd"):
        current_platform = "freeBSD"
    else:
        current_platform = _TOGA_PLATFORMS.get(sys.platform)
```

File: core/src/toga/platform.py [hunk 14]
```diff
@@ -21,6 +21,7 @@
     "darwin": "macOS",
     "ios": "iOS",
     "linux": "linux",
+    "freebsd": "freeBSD",
```
Comment (freakboy3742) [inline core/src/toga/platform.py:24]: This isn't needed if the `startswith("freebsd")` change is in place. The android version is technically redundant right now, but there's a possibility that we're going to get `sys.platform` to return `android` in future, so the line is there for that future possibility.

File: docs/how-to/contribute-code.rst (hunk 4 above)
Comment (freakboy3742) [reply docs/how-to/contribute-code.rst:34]: ...and I'm telling you that I've been working on this project for almost 8 years, and in *that entire time*, you're the first person who has asked for FreeBSD support. I haven't had *any* requests for other BSDs (or any other Unix, for that matter). 

I hate to break it to you, but non-Linux Unix platforms are *niche*. FWIW, Linux usage is pretty niche as well - but it's a *lot* more common than non-Linux Unix, and I'm not convinced that all Linux users actually know that they're running a "Unix-like" operating system. Many do - but you'd be surprised the sorts of questions we get from users. 

I'm also not wild about adding a second tab, since the instructions will be *exactly* the same, so you're adding a maintenance burden to the core team and all other contributors for the benefit of, so far, *exactly 1* user.

If FreeBSD users are going to start complaining that the best match for their usage instructions are listed under Linux, not "Unix", then I'm more than willing to offer them a 100% refund of their purchase price.

File: toga/setup.py [hunk 15]
```diff
@@ -9,6 +9,7 @@
         # Automatically installed platform backends
         ':sys_platform=="win32"': ["toga-winforms==%s" % version],
         ':sys_platform=="linux"': ["toga-gtk==%s" % version],
+        ':"freebsd" in sys_platform': ["toga-gtk==%s" % version],
```
Comment (freakboy3742) [inline toga/setup.py:12]: Does this actually work? Like - have you tested it? I didn't think the `sys_platform` syntax was "freeform Python". Can you provide a reference to docs that says this works, and/or another package that does this?

File: testbed/pyproject.toml [hunk 16]
```diff
@@ -63,6 +63,14 @@ flatpak_runtime = "org.gnome.Platform"
 flatpak_runtime_version = "42"
 flatpak_sdk = "org.gnome.Sdk"
 
+[tool.briefcase.app.testbed.freeBSD]
```
Comment (freakboy3742) [inline testbed/pyproject.toml:66]: This won't work until there's a freeBSD backend for Briefcase; and it wouldn't be needed anyway, because it exists to test *GTK*, not Linux.

File: docs/how-to/contribute-code.rst (hunk 5 above)
Comment (freakboy3742) [reply docs/how-to/contribute-code.rst:91]: I'm not I see sure you your previous comments address 2 and 3. Your other commits introduce compatibility with FreeBSD 12/14, but these docs still explicitly say 13 (point 2); and they're still at the *top* of the list of Unix options (point 3).

File: core/setup.cfg [hunk 17]
```diff
@@ -45,7 +45,9 @@ keywords =
     android
     windows
     winforms
+    unix-like
```
Comment (freakboy3742) [inline core/setup.cfg:48]: In case it wasn't clear - these "unix-like" keywords aren't needed. It's not a search term anyone will be looking for.

**Review by freakboy3742 — CHANGES_REQUESTED** [review]

File: core/setup.cfg (hunk 17 above)
Comment (MuhammadMouradG) [reply core/setup.cfg:48]: Sorry, I dropped the comments for 2 and 3, however the next commit will address them.

File: toga/setup.py (hunk 15 above)
Comment (MuhammadMouradG) [reply toga/setup.py:12]: Yes, it works. See [this](https://github.com/python-poetry/poetry/issues/2138#issuecomment-747130500) for example.

File: docs/how-to/contribute-code.rst (hunk 5 above)
Comment (MuhammadMouradG) [reply docs/how-to/contribute-code.rst]: Sorry, I dropped the comments for 2 and 3, however the next commit will address them.

**Comment by MuhammadMouradG** [comment]
As a side note to the point 4 that you mentioned in your comment, I tested `Toga` on a clean installation of `FreeBSD`. It comes with all required packages except that I mentioned in the `FreeBSD` section in the **docs**.

**Review by freakboy3742 — APPROVED** [review]
I've taken the opportunity to do a couple of unrelated tweaks to the installation instructions; but this all looks good. Thanks for the PR.
